api/src/tests/test_context.rs

```diff
@@ -229,6 +229,30 @@ impl TestContext {
         self.golden_output.as_ref().unwrap().log(&msg);
     }
 
+    pub fn check_golden_output_pruned(&mut self, msg: Value) {
+        self.check_golden_output(Self::prune_golden(msg))
+    }
+
+    /// Prune well-known excessively large entries from a resource array response.
+    /// TODO: we can't dump all resources of an account as golden output. As functionality
+    /// grows this becomes too much. Need a way to filter only the resources which folks want.
+    fn prune_golden(val: Value) -> Value {
```

Comment:
This kinda seems like the Golden files are being a pain if we're cutting these out.

Honestly it doesn't really matter all that much to me

Reply:
I think we should have a general concept in our REST APIs: *field masks*. These are quite common in REST Google APIs (disclaimer: I designed them personally 9 years ago so I might be biased, but it was a very common user request. [Here](https://developers.google.com/slides/api/guides/field-masks) is a gentle introduction.

Field masks is simply a set of field names which you use to prune a JSON resource object on server side (ideally they are passed down as deep as possible into to storage layer). Say you have resource with fields `x`, `y`, `z`. You can request this resource now and specify that you only want to see `x`. This applies to REST GET and REST LIST. It's easy to implement.

Reply:
Could we unify this with https://github.com/aptos-labs/aptos-core/blob/bcc1bd18937b7c4c69a4b47c64413a92adea7ded/api/src/tests/test_context.rs#L226